opengrok-indexer/src/main/java/org/opengrok/indexer/Metrics.java

```diff
@@ -53,90 +59,145 @@
-        }
+    private GraphiteConfig getGraphiteConfig(BaseGraphiteConfig baseGraphiteConfig) {
+        return new GraphiteConfig() {
+            @Override
+            public String get(String k) {
+                return null;
+            }
+
+            @Override
+            public int port() {
+                return baseGraphiteConfig.getPort();
+            }
+
+            @Override
+            public String host() {
+                return baseGraphiteConfig.getHost();
+            }
+
+            @Override
+            public GraphiteProtocol protocol() {
+                return baseGraphiteConfig.getProtocol();
+            }
+        };
+    }
 
-        @Override
-        public int port() {
-            return RuntimeEnvironment.getInstance().getStatsdConfig().getPort();
-        }
+    private PrometheusMeterRegistry prometheusRegistry;
 
-        @Override
-        public String host() {
-            return RuntimeEnvironment.getInstance().getStatsdConfig().getHost();
-        }
+    private MeterRegistry activeRegistry;
 
-        @Override
-        public boolean buffered() {
-            return true;
-        }
-    };
-
-    private static PrometheusMeterRegistry prometheusRegistry;
-    private static StatsdMeterRegistry statsdRegistry;
-
-    static {
-        MeterRegistry registry = null;
-
-        if (RuntimeEnvironment.getInstance().getStatsdConfig().isEnabled()) {
-            LOGGER.log(Level.INFO, "configuring StatsdRegistry");
-            statsdRegistry = new StatsdMeterRegistry(statsdConfig, Clock.SYSTEM);
-            registry = statsdRegistry;
-        } else if (!RuntimeEnvironment.getInstance().isIndexer()) {
-            LOGGER.log(Level.INFO, "configuring PrometheusRegistry");
-            prometheusRegistry = new PrometheusMeterRegistry(PrometheusConfig.DEFAULT);
-            registry = prometheusRegistry;
-        }
+    private static final Metrics instance = new Metrics();
 
-        if (registry != null) {
-            new ClassLoaderMetrics().bindTo(registry);
-            new JvmMemoryMetrics().bindTo(registry);
-            new JvmGcMetrics().bindTo(registry);
-            new ProcessorMetrics().bindTo(registry);
-            new JvmThreadMetrics().bindTo(registry);
-        }
+    private Metrics() {
     }
 
-    private Metrics() {
+    /**
+     * @return the only instance of Metrics
+     */
+    public static Metrics getInstance() {
+        return instance;
     }
 
-    public static void updateSubFiles(List<String> subFiles) {
+    /**
+     * Configure meter registry.
+     * @param type type of meter registry
+     */
+    public void configure(MeterRegistryType type) {
+        switch (type) {
+            case PROMETHEUS:
+                LOGGER.log(Level.INFO, "configuring Prometheus registry with default config");
+                prometheusRegistry = new PrometheusMeterRegistry(PrometheusConfig.DEFAULT);
```

Comment:
If this method is called a second time with a different `type`, might not a previously existing, non-null `prometheusRegistry` be orphaned?

Reply:
I realize this is not a good design however makes testing easier in `MetricsConfigureTest`. Throw an exception as an alternative (or maybe silently return and document that) and split the test class ?